test/functional/terminal/ex_terminal_spec.lua

```diff
@@ -68,6 +68,11 @@ describe(':terminal (with fake shell)', function()
     execute("terminal "..(cmd and cmd or ""))
   end
 
+  local function terminal_with_fake_shell_with_arguments(cmd)
+    nvim('set_option', 'shell', nvim_dir..'/shell-test -t jeff')
```

Comment:
would prefer to put this explicitly in the tests, then call the normal `terminal_with_fake_shell()`.